frontend/src/pages/Admin/EnterPhoneNumber/Dialpad/index.tsx

```diff
@@ -26,20 +24,37 @@ const Dialpad = () => {
     navigateNextPage,
   } = useDialPad();
 
-  const { status: customerStatus } = useQuery<CustomerPhoneNumberRes>(['customer', phoneNumber], {
+  const {
+    data: customers,
+    status: customerStatus,
+    refetch: refetchCustomers,
+  } = useQuery<CustomerPhoneNumberRes>(['customer', phoneNumber], {
     queryFn: () => getCustomer({ params: { phoneNumber: phoneNumber.replaceAll('-', '') } }),
     onSuccess: (data) => {
       if (data.customer.length === 0) {
         openModal();
         return;
       }
-      navigate(ROUTER_PATH.selectCoupon, {
-        state: { phoneNumber: phoneNumber.replaceAll('-', '') },
-      });
+      navigateNextPage(data.customer[0]);
     },
     enabled: isDone,
   });
 
+  const requestTemporaryCustomer = () => {
+    mutateTemporaryCustomer({ body: { phoneNumber: phoneNumber.replaceAll('-', '') } });
+  };
+
+  const { mutate: mutateTemporaryCustomer } = useMutation({
+    mutationFn: postTemporaryCustomer,
+    onSuccess: async () => {
+      await refetchCustomers();
+      if (customers?.customer[0]) navigateNextPage(customers.customer[0]);
+    },
```

Comment:
저도 최근에 안건데 refetch 대신 invaildate Queries 를 사용하는 것이 더 좋아보여요! refetch같은 경우 query가 observing 할 수 없다고 하네요~!

https://github.com/TanStack/query/discussions/2468
https://tanstack.com/query/v4/docs/react/guides/invalidations-from-mutations

Reply:
추가 하자면, 다음과 같이 변경할 수 있을 것 같아요. 쿼리를 무효화한다고 동기적으로 알려줄 수 있고, 이 후 무효화에 관한 refetch는 query가 알아서 해줄 듯 합니다!!
```suggestion
  const { mutate: mutateTemporaryCustomer } = useMutation({
    mutationFn: postTemporaryCustomer,
    onSuccess: () => {
      queryClient.invalidateQueries({ queryKey: ['customer'] }); // 위에서 useQueryClient의 선언 필요
      if (customers?.customer[0]) navigateNextPage(customers.customer[0]);
    },
```

Reply:
아 새로 요청하기보다 이전의 쿼리를 무효화한다고 알려주는거군요! 수정하겠습니다~